### ISSUE TITLE
application/xml representation

### ISSUE THREAD

**Issue by brianchenry** [JakeGinnivan/WebApi.Hal#112]
I just recently pulled the source for WebApi.Hal, including the sample Beers application.  I was quickly able to run and test the application by supplying either a application/hal+json or application/hal+xml Accept header.  If no header is supplied it returns json by default with null hypermedia.  If I supply an application/xml Accept header it still returns json.

How can I get it to return an xml format with null hypermedia for an application/xml Accept header?

In short, the standard xml representation does not work.

Thanks,
Brian

**Comment by kbrichan** [comment]
The sample outputs ordinary JSON because the default .Net JSON serializer doesn't suffer from the same requirements as the default .Net xml serializer which requires POCOs to be marked as serializable or to enable data contract serialization.

The beer sample POCOs aren't set up for the default .Net xml serializer. The sample was for testing and demonstrating HAL, not ordinary xml.

However, if you want the sample to output ordinary xml, put this in the global.asax.cs where it is registering the media formatters:

GlobalConfiguration.Configuration.Formatters.XmlFormatter.UseDataContractSerializer = true;

**Comment by brianchenry** [comment]
Brook,

Thank you for your prompt response.  I did try the solution you proposed and it did not work, it still returns json.  The only thing that I've come up with so far is to add this SupportedMediaTypes.Add(new MediaTypeHeaderValue("application/xml")); to the XmlHalMediaTypeFormatter constructor which essentially then behaves identically to the hal+xml representation.

I would still like to figure out how to get the plain xml representation to work however.  I can see a use case where a legacy consumer would want to use my API to consume xml.  Further, I see benefit in being able to turn hal on\off from the consumer's perspective to reduce the payload in a production environment.

Any advice is appreciated.

**Comment by kbrichan** [comment]
I totally agree the client should be able to specify their desired format, and do so using the accept header. Setting the UseDataContractSerializer true is a suggestion I found in searching for how to enable xml in .net webapi. It seems others are having the exact same problem with webapi in general.

I found the following two links, which you might consult to see if it gives you some assistance. Once you've figured it out, please post back here and we can change the beer sample accordingly.

http://forums.asp.net/t/1787054.aspx?XML+

http://forums.asp.net/t/1804419.aspx?how+to+set+a+Web+API+REST+service+to+always+return+XML+not+JSON

**Comment by kbrichan** [comment]
By the way, both of those links suggest to set UseDataContractSerializer true, and they both say it worked. I'd be curious to know, once you figure it out, why doing this in the beer sample fails.

**Comment by panmanphil** [comment]
closing stale issue
